### ISSUE TITLE
stake-pool: preferred validator can be set to validator not part of the pool

### ISSUE THREAD

**Issue by joncinque** [solana-labs/solana-program-library#2262]
#### Problem

The preferred withdraw / deposit vote address can be set to a validator that is not part of the pool, by 1) removing the validator from the pool (state of the entry is now ReadyForRemoval) 2) changing the preferred address to that validator 3) calling CleanupRemovedValidatorEntries only after 2.

#### Solution

Avoid setting the preferred validator if it's marked as `ReadyForRemoval`.
